js/NetworkPane.js

```diff
@@ -0,0 +1,221 @@
+import React from 'react';
+
+const Pane = require('./Pane');
+
+class NetworkPane extends React.Component {
+  _paneRef = null;
+  _networkRef = null;
+  _width = null;
+  _height = null;
+
+  handleDownload = () => {
+    var link = document.createElement('a');
+    link.download = `${this.props.title || 'visdom_image'}.jpg`;
+    link.href = this.props.content.src;
+    link.click();
+  };
+
+  CreateNetwork = graph => {
+    var width = 500,
+      height = 500;
+    var color = d3.scale.category10();
+    var force = d3.layout
+      .force()
+      .charge(-120)
+      .linkDistance(120)
+      .size([width, height]);
+    var svg = d3.select('.Network_Div').select('svg');
+    if (svg.empty()) {
+      svg = d3
+        .select('.Network_Div')
+        .append('svg')
+        .attr('width', width)
+        .attr('height', height);
+    }
+
+    if (this.props.Gtype == 'dgph' || this.props.Gtype == 'dwgph') {
```

Comment:
`Gtype` string can be easily substituted by the boolean `directed` key in `opts` dict.
The execution pipeline enters this block of code iff the graph is directed.

Reply:
Thanks for this suggestion @dizcza, I think it would be best to have this in opts.